python_modules/libraries/dagster-openai/dagster_openai/resources.py

```diff
@@ -0,0 +1,254 @@
+from collections import defaultdict
+from contextlib import contextmanager
+from enum import Enum
+from functools import wraps
+from typing import Generator, List, Optional
+from weakref import WeakKeyDictionary
+
+from dagster import AssetExecutionContext, ConfigurableResource, InitResourceContext
+from openai import Client
+from pydantic import Field, PrivateAttr
+
+
+class ApiResourceEnum(Enum):
```

Comment:
> Naming quibble - since this name overloads the Dagster "Resource" noun, it could be pretty confusing to users (they construct an OpenAI Resource & pass it API resources). Are these called resources in OpenAI docs?

I agree, but indeed, they are called resources in the OpenAI library we are wrapping. I also thought of ApiEndpointClassesEnum to make it obvious that it is not related to a dagster resource. I can make the change.

Reply:
`ApiEndpointClassesEnum` sounds not bad to me - i'd lean towards very descriptive to start with.